docs(zipkin-oss) describe spans, tags and annotations for zipkin 1.3 & 1.4 (#3021)
Kong/docs.konghq.com: app/_hub/kong-inc/zipkin/index.md

The `http_endpoint` configuration variable must contain the full uri including scheme, host, port and path sections (i.e. your uri likely ends in `/api/v2/spans`).

### Spans

The plugin does *request sampling*. For each request which triggers the plugin, a random number between 0 and 1 is chosen.

If the number is greater than the configured `sample_ratio`, then a trace with several spans will be generated. If `sample_ratio` is set to 1, then all requests will generate a trace (this might be very noisy).

For each request that gets traced, the following spans are produced:

> * **Request Span**: 1 per request. Encompasses the whole request in kong (kind: SERVER).

Reviewer: ```suggestion
* **Request span**: 1 per request. Encompasses the whole request in kong (kind: SERVER).
```

>   The Proxy Span and Balancer Spans are children of this span. It Contains logs/annotations for the rewrite phase:

Reviewer: ```suggestion
  The Proxy and Balancer spans are children of this span. It contains the following logs/annotations for the rewrite phase:
```

  * `krs` - `kong.rewrite.start`
  * `krf` - `kong.rewrite.finish`

>   The Request Span has the following tags:

Reviewer: ```suggestion
  The Request span has the following tags:
```

>   * `lc`: Hardcoded to `kong`

Reviewer: ```suggestion
  * `lc`: Hardcoded to `kong`.
```

>   * `kong.service`: The uuid of the Service matched when processing the request, if any

Reviewer: ```suggestion
  * `kong.service`: The uuid of the service matched when processing the request, if any.
```

>   * `kong.service_name`: The name of the Service matched when processing the request, if said Service exists and has a `name` attribute

Reviewer: ```suggestion
  * `kong.service_name`: The name of the service matched when processing the request, if service exists and has a `name` attribute.
```

>   * `kong.route`: The uuid of the Route matched when processing the request, if any (it can be nil on non-matched requests)

Reviewer: ```suggestion
  * `kong.route`: The uuid of the route matched when processing the request, if any (it can be nil on non-matched requests).
```

>   * `kong.route_name`: The name of the Route matched when processing the request, if said Route exists and has a `name` attribute

Reviewer: ```suggestion
  * `kong.route_name`: The name of the route matched when processing the request, if route exists and has a `name` attribute.
```

>   * `http.method`: The HTTP method used on the original request (only for HTTP requests)

Reviewer: ```suggestion
  * `http.method`: The HTTP method used on the original request (only for HTTP requests).
```

>   * `http.path`: The path of the request (only for HTTP requests)

Reviewer: ```suggestion
  * `http.path`: The path of the request (only for HTTP requests).
```

>   * If the plugin `tags_header` config option is set, and the request contains headers with the appropiate name and encoding tags correctly, then the trace will include said tags

Reviewer: ```suggestion
  * If the plugin `tags_header` config option is set, and the request contains headers with the appropriate name and correct encoding tags, then the trace will include the tags.
```

>   * If the plugin `static_tags` config option is set, then the tags in said config option will be included on the

Reviewer: ```suggestion
  * If the plugin `static_tags` config option is set, then the tags in the config option will be included in the trace.
```

> * **Proxy span**: 1 per request. Encompassing most of Kong's internal processing of a request (kind: CLIENT)

Reviewer: ```suggestion
* **Proxy span**: 1 per request, encompassing most of Kong's internal processing of a request (kind: CLIENT).
```

>   Contains logs/annotations for the rest start/finish of the of the Kong plugin phases:

Reviewer: ```suggestion
  Contains the following logs/annotations for the rest start/finish of the of the Kong plugin phases:
```

Reviewer: I'm not sure how "rest" fits in this sentence?

  * `kas` - `kong.access.start`
  * `kaf` - `kong.access.finish`
  * `kbs` - `kong.body_filter.start`
  * `kbf` - `kong.body_filter.finish`
  * `khs` - `kong.header_filter.start`
  * `khf` - `kong.header_filter.finish`
  * `kps` - `kong.preread.start` (only for stream requests)
  * `kpf` - `kong.preread.finish` (only for stream requests)

> * **Balancer span(s)**: 0 or more per request, each encompassing one balancer attempt (kind: CLIENT)

Reviewer: ```suggestion
* **Balancer span(s)**: 0 or more per request, each encompassing one balancer attempt (kind: CLIENT).
```

> Contains tags specific to the load balancing:

Reviewer: ```suggestion
Contains the following tags specific to load balancing:
```

>   * `kong.balancer.try`: A number indicating the attempt (1 for the first load-balancing attempt, 2 for the second, and so on)

Reviewer: ```suggestion
  * `kong.balancer.try`: A number indicating the attempt (1 for the first load-balancing attempt, 2 for the second, and so on).
```

>   * `peer.ipv4` or `peer.ipv6` for the balancer ip

Reviewer: ```suggestion
  * `peer.ipv4` or `peer.ipv6` for the balancer IP.
```

>   * `peer.port` for the balanced port

Reviewer: ```suggestion
  * `peer.port` for the balanced port.
```

>   * `error`: Set to `true` if the balancing attempt was unsuccessful. Otherwise unset

Reviewer: ```suggestion
  * `error`: Set to `true` if the balancing attempt was unsuccessful, otherwise unset.
```

Reviewer: Does this mean it is set to `false` by default?

>   * `http.status_code`: the HTTP status code received, in case of error

Reviewer: ```suggestion
  * `http.status_code`: The HTTP status code received, in case of error.
```

>   * `kong.balancer.state`: an nginx-specific description of the error: `next/failed` for HTTP failures, `0` for stream failures.

Reviewer: ```suggestion
  * `kong.balancer.state`: An NGINX-specific description of the error, `next/failed` for HTTP failures, or `0` for stream failures.
```

>      Equivalent to `state_name` in OpenResty's Balancer's `get_last_failure` function.

Reviewer: ```suggestion
     Equivalent to `state_name` in OpenResty's balancer's `get_last_failure` function.
```
